### PR TITLE
Improve docs of leaky_relu and relu

### PR DESCRIPTION
**Pull request by keisuke-umezawa**
This is PR for improving docs of functions/links. Related issue: #2182

Modified functions/links:
- leaky_relu
- relu

### CODE REVIEW

File: chainer/functions/activation/leaky_relu.py [hunk 1]
```diff
@@ -43,15 +43,31 @@ def backward_gpu(self, x, gy):
...
-    is a configurable slope value.
+    .. math:: f(x)=\\max(x, ax),
+
+    where :math:`a` is a configurable slope value.
 
     Args:
-        x (~chainer.Variable): Input variable.
+        x (:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+        :class:`cupy.ndarray`):
+            Input variable. A :math:`(s_1, s_2, ..., s_n)`-shaped float array.
```
Comment (mitmul) [inline chainer/functions/activation/leaky_relu.py]: For consistency, I would say it's better to replace `s_n` with `s_N`

File: chainer/functions/activation/leaky_relu.py [hunk 2]
```diff
@@ -43,15 +43,31 @@ def backward_gpu(self, x, gy):
...
         slope (float): Slope value :math:`a`.
 
     Returns:
-        ~chainer.Variable: Output variable.
+        ~chainer.Variable: Output variable. A
+        :math:`(s_1, s_2, ..., s_n)`-shaped float array.
```
Comment (mitmul) [inline chainer/functions/activation/leaky_relu.py]: Ditto.

File: chainer/functions/activation/leaky_relu.py [hunk 3]
```diff
@@ -43,15 +43,31 @@ def backward_gpu(self, x, gy):
...
+
+    .. admonition:: Example
+
+        >>> x = np.array([[-1, 0], [2, -3], [-2, 1]], 'f')
+        >>> x
+        array([[-1.,  0.],
+               [ 2., -3.],
+               [-2.,  1.]], dtype=float32)
+        >>> F.leaky_relu(x).data
```
Comment (mitmul) [inline chainer/functions/activation/leaky_relu.py]: How about specifying `slope=0.2` explicitly here to show the results surely use `a = 0.2` in the above equation?

Comment (keisuke-umezawa) [inline chainer/functions/activation/leaky_relu.py]: ok! I will write the parameter explicitly.

File: chainer/functions/activation/leaky_relu.py (hunk 1 above)
Comment (keisuke-umezawa) [inline chainer/functions/activation/leaky_relu.py]: I also fixed the relu file because it had the same problem.

**Comment by keisuke-umezawa** [comment]
I fixed the docstrings as mentioned above. Please merge after the test is passed.

File: chainer/functions/activation/leaky_relu.py [hunk 4]
```diff
@@ -43,15 +43,31 @@ def backward_gpu(self, x, gy):
 def leaky_relu(x, slope=0.2):
     """Leaky Rectified Linear Unit function.
 
-    This function is expressed as :math:`f(x) = \\max(x, ax)`, where :math:`a`
```
Comment (mitmul) [inline chainer/functions/activation/leaky_relu.py]: Is the first sentence necessarily removed? I think it's OK to be here.

Comment (keisuke-umezawa) [inline chainer/functions/activation/leaky_relu.py]: Ok. revert the first sentence.

**Comment by keisuke-umezawa** [comment]
@mitmul I followed the review and rebased the branch against master. Please rerun the Jenkins test.

**Comment by mitmul** [comment]
LGTM

**Pull request merged**